src/Microsoft.DotNet.Interactive.ExtensionLab/MsSqlServiceClient.cs

```diff
@@ -84,58 +84,69 @@ public async Task<bool> DisconnectAsync(string ownerUri)
                     await writer.WriteLineAsync(command.Code);
                 }
+                await SendTextChangeNotification(fileUri, command.Code, oldFileContents);
+            }
 
-                var fileUri = GetUriForFilePath(filePath);
-                TextDocumentIdentifier docId = new TextDocumentIdentifier() { Uri = fileUri };
-                Position position = new Position() { Line = command.LinePosition.Line, Character = command.LinePosition.Character };
-                CompletionContext context = new CompletionContext() { TriggerKind = (int)CompletionTriggerKind.Invoked };
-                var completionParams = new CompletionParams() { TextDocument = docId, Position = position, Context = context };
-                var sqlCompletionItems = await rpc.InvokeWithParameterObjectAsync<SqlCompletionItem[]>("textDocument/completion", completionParams);
+            TextDocumentIdentifier docId = new TextDocumentIdentifier() { Uri = fileUri };
+            Position position = new Position() { Line = command.LinePosition.Line, Character = command.LinePosition.Character };
+            CompletionContext context = new CompletionContext() { TriggerKind = (int)CompletionTriggerKind.Invoked };
+            var completionParams = new CompletionParams() { TextDocument = docId, Position = position, Context = context };
 
-                return sqlCompletionItems.Select(item =>
-                {
-                    return new CompletionItem(
-                        item.Label,
-                        item.Kind != null ? Enum.GetName(typeof(SqlCompletionItemKind), item.Kind) : string.Empty,
-                        item.FilterText,
-                        item.SortText,
-                        item.InsertText,
-                        item.Documentation);
-                });
-            }
-            finally
+            var sqlCompletionItems = await rpc.InvokeWithParameterObjectAsync<SqlCompletionItem[]>("textDocument/completion", completionParams);
+
+            return sqlCompletionItems.Select(item =>
             {
-                File.Delete(filePath);
-            }
+                return new CompletionItem(
+                    item.Label,
+                    item.Kind != null ? Enum.GetName(typeof(SqlCompletionItemKind), item.Kind) : string.Empty,
+                    item.FilterText,
+                    item.SortText,
+                    item.InsertText,
+                    item.Documentation);
+            });
         }
 
         public async Task<Hover> ProvideHoverAsync(string filePath, RequestHoverText command)
         {
-            var tempFileStream = File.Open(filePath, FileMode.OpenOrCreate);
-            try
+            string oldFileContents;
+            using (var tempFileStream = File.Open(filePath, FileMode.OpenOrCreate))
+            using (var reader = new StreamReader(tempFileStream))
+            {
+                oldFileContents = await reader.ReadToEndAsync();
+            }
```

Comment:
You can potentially simplify using `File.ReadAllText`.

Reply:
This part also needs to create the temp file if it doesn't exist.

Reply:
Disregard that, looks like GetTempFileName also creates the file.